Improvements to the code generated by GadToolsBox contributed by Kamil Iskra
(iskra@student.uci.agh.edu.pl).

*** ixprefs.c.gtb	Sun Jun 25 12:36:45 1995
--- ixprefs.c	Sun Jun 25 12:50:28 1995
***************
*** 14,32 ****
  #include <libraries/gadtools.h>
  #include <graphics/displayinfo.h>
  #include <graphics/gfxbase.h>
- #include <clib/exec_protos.h>
- #include <clib/intuition_protos.h>
- #include <clib/gadtools_protos.h>
- #include <clib/graphics_protos.h>
- #include <clib/utility_protos.h>
  #include <string.h>
- #include <clib/diskfont_protos.h>
  
! #include <pragmas/exec_pragmas.h>
! #include <pragmas/intuition_pragmas.h>
! #include <pragmas/gadtools_pragmas.h>
! #include <pragmas/graphics_pragmas.h>
! #include <pragmas/utility_pragmas.h>
  
  #include "ixprefs.h"
  
--- 14,26 ----
  #include <libraries/gadtools.h>
  #include <graphics/displayinfo.h>
  #include <graphics/gfxbase.h>
  #include <string.h>
  
! #include <proto/exec.h>
! #include <proto/intuition.h>
! #include <proto/gadtools.h>
! #include <proto/graphics.h>
! #include <proto/utility.h>
  
  #include "ixprefs.h"
  
***************
*** 103,125 ****
  };
  
  ULONG ixprefsGTags[] = {
  	(TAG_DONE),
  	(TAG_DONE),
  	(TAG_DONE),
! 	(TAG_DONE),
! 	(TAG_DONE),
! 	(TAG_DONE),
! 	(TAG_DONE),
  	(GTIN_Number), 0, (GTIN_MaxChars), 10, (TAG_DONE),
  	(GTIN_Number), 0, (GTIN_MaxChars), 10, (TAG_DONE),
! 	(TAG_DONE),
  	(GTIN_Number), 0, (GTIN_MaxChars), 10, (TAG_DONE),
! 	(TAG_DONE),
! 	(TAG_DONE),
! 	(TAG_DONE),
! 	(TAG_DONE),
! 	(TAG_DONE),
! 	(TAG_DONE)
  };
  
  static UWORD ComputeX( UWORD value )
--- 97,119 ----
  };
  
  ULONG ixprefsGTags[] = {
+ 	(GTCB_Scaled), TRUE, (TAG_DONE),
  	(TAG_DONE),
  	(TAG_DONE),
  	(TAG_DONE),
! 	(GTCB_Scaled), TRUE, (TAG_DONE),
! 	(GTCB_Scaled), TRUE, (TAG_DONE),
! 	(GTCB_Scaled), TRUE, (TAG_DONE),
  	(GTIN_Number), 0, (GTIN_MaxChars), 10, (TAG_DONE),
  	(GTIN_Number), 0, (GTIN_MaxChars), 10, (TAG_DONE),
! 	(GTCB_Scaled), TRUE, (TAG_DONE),
  	(GTIN_Number), 0, (GTIN_MaxChars), 10, (TAG_DONE),
! 	(GTCB_Scaled), TRUE, (TAG_DONE),
! 	(GTCB_Scaled), TRUE, (TAG_DONE),
! 	(GTCB_Scaled), TRUE, (TAG_DONE),
! 	(GTCB_Scaled), TRUE, (TAG_DONE),
! 	(GTCB_Scaled), TRUE, (TAG_DONE),
! 	(GTCB_Scaled), TRUE, (TAG_DONE)
  };
  
  static UWORD ComputeX( UWORD value )
***************
*** 240,246 ****
  	if (( wleft + ww + OffX + Scr->WBorRight ) > Scr->Width ) wleft = Scr->Width - ww;
  	if (( wtop + wh + OffY + Scr->WBorBottom ) > Scr->Height ) wtop = Scr->Height - wh;
  
! 	if ( ! ( ixprefsFont = OpenDiskFont( Font )))
  		return( 5L );
  
  	if ( ! ( g = CreateContext( &ixprefsGList )))
--- 234,240 ----
  	if (( wleft + ww + OffX + Scr->WBorRight ) > Scr->Width ) wleft = Scr->Width - ww;
  	if (( wtop + wh + OffY + Scr->WBorBottom ) > Scr->Height ) wtop = Scr->Height - wh;
  
! 	if ( ! ( ixprefsFont = OpenFont( Font )))
  		return( 5L );
  
  	if ( ! ( g = CreateContext( &ixprefsGList )))
***************
*** 266,275 ****
  			return( 2L );
  	}
  
! 	if ( ! ( ixprefsMenus = CreateMenus( ixprefsNewMenu, GTMN_FrontPen, 0L, TAG_DONE )))
  		return( 3L );
  
! 	LayoutMenus( ixprefsMenus, VisualInfo, TAG_DONE );
  
  	if ( ! ( ixprefsWnd = OpenWindowTags( NULL,
  				WA_Left,	wleft,
--- 260,269 ----
  			return( 2L );
  	}
  
! 	if ( ! ( ixprefsMenus = CreateMenusA( ixprefsNewMenu, NULL )))
  		return( 3L );
  
! 	LayoutMenus( ixprefsMenus, VisualInfo, GTMN_NewLookMenus, TRUE, TAG_DONE );
  
  	if ( ! ( ixprefsWnd = OpenWindowTags( NULL,
  				WA_Left,	wleft,
***************
*** 284,289 ****
--- 278,284 ----
  				WA_PubScreen,	Scr,
  				WA_AutoAdjust,	TRUE,
  				WA_PubScreenFallBack,	TRUE,
+ 				WA_NewLookMenus,	TRUE,
  				TAG_DONE )))
  	return( 4L );
  
